python-package/lightgbm/dask.py

```diff
@@ -126,6 +126,22 @@ def _find_ports_for_workers(client: Client, worker_addresses: Iterable[str], loc
     return worker_ip_to_port
 
 
+def _find_random_open_port() -> int:
+    """Find a random open port on the machine.
+
+    Returns
+    -------
+    port : int
+        A free port on the machine.
+    """
+    import socket
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
Remove this line. `socket` is already imported in appropriate place
https://github.com/microsoft/LightGBM/blob/c3ac77b570b687e948d63b2078adcf4c37f3dec5/python-package/lightgbm/dask.py#L9

```suggestion
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Reply:
I did this on purpose because I thought `client.run` would fail without importing the module inside the function but it works, so thank you for pointing it out. I'll remove it in my next push.